### PR TITLE
user - personal details page

### PR DESCRIPTION
**Pull request by MartaSolo**
User personal details page with updating user data, styling for desktop and mobile.
personal_details_desktop
personal_details_errors_desktop
personal_details_success_desktop
personal_details_success_mobile

### CODE REVIEW

File: src/components/Global/BaseInput.vue [hunk 1]
```diff
@@ -100,15 +100,22 @@ export default defineComponent({
 .input__input.register__name,
 .input__input.register__surname,
 .input__input.login__email,
-.input__input.login__password {
+.input__input.login__password,
+.input__input.account__email,
+.input__input.account__name,
+.input__input.account__surname,
+.input__input.account__password,
+.input__input.account__password-new {
```
Comment (szymonlicau) [inline src/components/Global/BaseInput.vue]: Won't this be most of the inputs anyway then? Wouldn't it make sense to always add this margin and error case handling?

File: src/components/user/AccountTabs.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      class="account__tab"
+      :class="{ active: activeTab === 'reservations' }"
+      @click="changeTab('reservations')"
+      >Reservations</BaseButton
+    >
+  </section>
+</template>
+
+<style lang="scss" scoped>
+.account_tabs {
```
Comment (szymonlicau) [inline src/components/user/AccountTabs.vue]: We should stick to kebab-case in our normal classes 😄 

```suggestion
.account-tabs {
```

File: src/components/user/PersonalDetails.vue [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+  )
+    isValid = true;
+  if (birthdayTouched.value && birthdayError() === "error") isValid = false;
+  if (
+    (showNewPassword.value && !updatedUser.value.newPassword) ||
+    (showNewPassword.value && newPasswordError())
+  )
+    isValid = false;
+  return isValid;
+};
```
Comment (szymonlicau) [inline src/components/user/PersonalDetails.vue]: Shouldn't all of these just be computed properties? This way we do not need to call them and always have their value up to date

File: src/components/user/AccountTabs.vue [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import BaseButton from "@/components/global/BaseButton.vue";
+
+const props = withDefaults(
+  defineProps<{
+    activeTab: string;
```
Comment (szymonlicau) [inline src/components/user/AccountTabs.vue]: When we know it's a strict list of known values, we should definitely provide hints on that and restrict our props as well.
```suggestion
    activeTab: 'personalDetails' | 'reservations';
```

**Review by szymonlicau — CHANGES_REQUESTED** [review]

File: src/components/Global/BaseInput.vue (hunk 1 above)
Comment (MartaSolo) [reply src/components/Global/BaseInput.vue]: Yes, it makes sense :)

File: src/components/user/PersonalDetails.vue (hunk 3 above)
Comment (MartaSolo) [reply src/components/user/PersonalDetails.vue]: At some point I had some problem with computed and decided to go with functions, but now I corrected and everything works fine :)

**Review by szymonlicau — APPROVED** [review]

**Pull request merged**